tools/benchmark/api-report/benchmark.alpha.api.md

```diff
@@ -172,6 +172,14 @@ export interface MochaExclusiveOptions {
 // @public
 export interface OnBatch {
     beforeEachBatch?: () => void;
+    beforeEachBatchAsync?: never;
+}
+
+// @public
+export interface OnBatchAsync {
+    // @deprecated
+    beforeEachBatch?: () => void;
+    beforeEachBatchAsync?: () => Promise<void>;
```

Comment:
This change seems unnecessary complex. While we have to make the actual function being timed clearly sync or async to avoid the over head of checking which each iteration, I don't see an issue with making `beforeEachBatch?: () => void;` just typed as `beforeEachBatch?: () => void | promise<unknown>;` then just awaiting what ever it returns unconditionally.

That should keep the API identical, except that async beforeEachBatch will work. This is similar to how mocha deals with hooks: same API for sync and async.

Reply:
That's how I started.  I was bothered by the lack of compiler error when passing args with synchronous benchmarkFn by async beforeEachBatch

Reply:
Looking at the implementation, I guess that would be an issue with how a few things are currently factored, like `runBenchmarkSync`, so some of this refactoring might be needed to support async beforeEachBatch.